### PR TITLE
Remove Duplicates from defaults.yaml and map.jinja

### PR DESCRIPTION
**Pull request by puneetk**

### CODE REVIEW

**Comment by gravyboat** [comment]
Hmm, are we sure that we want to remove all those defaults? What if someone wants to modify it in the future?

**Comment by puneetk** [comment]
They would add it to the map.jinja for the over rides they need ?

**Comment by gravyboat** [comment]
Yeah I just mean removing the existing examples that are in there since this pulls it out. I'm all for it, just something to consider.

**Comment by aboe76** [comment]
@puneetk did you test this with pillar salt:lookup still works?

**Comment by puneetk** [comment]
nope, but should work, assuming functions work as designed and to my understanding. I have used similar stuff in other places.

File: salt/map.jinja [hunk 1]
```diff
@@ -1,72 +1,60 @@
...
+  , merge=salt['pillar.get']('salt:lookup'))
+%}
 
-{% set salt_settings = salt['pillar.get']('salt', default=default_settings.salt,
-  merge=True) %}
+{## Merge the flavor_map to the default settings ##}
+{% do default_settings.salt.update(os_family_map) %}
+    
+{## Merge in template:lookup pillar ##}
+{% set template_settings = salt['pillar.get'](
```
Comment (aboe76) [inline salt/map.jinja]: this should be salt_settings instead of template_settings

**Comment by aboe76** [comment]
@puneetk thanks, for updating the map.jinja I have tested it and I think it is good to go,

@gravyboat do you want to test also?

**Comment by puneetk** [comment]
@forrest wants to setup travis for this repo, i second that motion.

**Comment by aboe76** [comment]
@puneetk and @gravyboat I thinks that's a very good idea, especially because this formula is heavily in use and is something of a calling card for saltstack if it can manage itself in a sane way.

**Comment by gravyboat** [comment]
@aboe76 What idea are you saying is a good idea? Sorry.

**Comment by aboe76** [comment]
ow @gravyboat I thought @puneetk was talking about you setting up travis ci for this?
if that's not the case nevermind.

**Comment by gravyboat** [comment]
@aboe76 It is, I didn't see @puneetk's last message for some reason when I refreshed. Probably scrolled down too much.

**Comment by aboe76** [comment]
@gravyboat and @puneetk I have tested the pull request against archlinux machine wit salt:lookup pillar data and an opensuse system, no errors found, so I'm happy with the pull request.

if @gravyboat wants to setup travis first in order to test this, I'm ok with it.

**Comment by gravyboat** [comment]
I'm going to merge this. Don't have time to work on the travis file right now.

**Pull request merged**